btree.go

```diff
@@ -64,6 +64,42 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
```

Comment:
Add note that two btrees using the same freelist are not safe for concurrent write access.